packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -123,13 +139,35 @@ async function createWalletAddress(
     return WalletAddressError.InvalidUrl
   }
   try {
-    return await WalletAddress.query(deps.knex)
+    const wallet = await WalletAddress.query(deps.knex)
       .insertAndFetch({
         url: options.url,
         publicName: options.publicName,
         assetId: options.assetId
       })
       .withGraphFetched('asset')
+    let addProperties = options.additionalProperties
+    if (addProperties) {
+      // remove blank key/value pairs:
+      addProperties = addProperties.filter((itm) => {
+        return !(
+          itm.fieldKey.trim().length == 0 || itm.fieldValue.trim().length == 0
+        )
+      })
+      // set defaults:
+      const now = new Date()
+      for (const prop of addProperties) {
+        prop.walletAddressId = wallet.id
+        prop.createdAt = now
+        prop.updatedAt = now
+      }
```

Comment:
I guess we'd still need to set the `walletAddressId`... Could do that inside the iteration above (although it would need to be changed to a for loop that conditionally pushes to `addProperties`) but I guess these arrays are never going to be very big so maybe this isn't really a big deal to be honest.